src/industry_gui.cpp

```diff
@@ -1209,12 +1336,19 @@ class IndustryDirectoryWindow : public Window {
 
 			this->industries.shrink_to_fit();
 			this->industries.RebuildDone();
-			this->vscroll->SetCount((uint)this->industries.size()); // Update scrollbar as well.
 		}
 
-		if (!this->industries.Sort()) return;
 		IndustryDirectoryWindow::last_industry = nullptr; // Reset name sorter sort cache
-		this->SetWidgetDirty(WID_ID_INDUSTRY_LIST); // Set the modified widget dirty
+
+		auto filter = std::make_pair(this->cargo_filter[this->accepted_cargo_filter_criteria],
+		                             this->cargo_filter[this->produced_cargo_filter_criteria]);
+
+		this->industries.Filter(filter);
+		this->industries.Sort();
+
+		this->vscroll->SetCount(this->industries.size()); // Update scrollbar as well.
```

Comment:
New warning on MSVC here: `##[warning]src\industry_gui.cpp(1349,0): Warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data` - SetCount (currently) takes an int